### PR TITLE
Origin/bicaws 1585 adding user service messages on error and login pages

### PR DESCRIPTION
**Pull request by MihaiMadeT**

### CODE REVIEW

File: src/pages/faq.tsx [hunk 1]
```diff
@@ -18,50 +23,77 @@ interface faqElement {
...
+export const getServerSideProps = async () => {
+  const connection = getConnection()
+  const serviceMessagesResult = await getServiceMessages(connection, 0)
+  let serviceMessages: ServiceMessage[] = []
+  if (isError(serviceMessagesResult)) {
+    logger.error(serviceMessagesResult)
+  } else {
+    serviceMessages = JSON.parse(JSON.stringify(serviceMessagesResult.result))
+  }
+  console.log(serviceMessages)
```
Comment (daviesjamie) [inline src/pages/faq.tsx:40]: Do we need to be logging out to the console here?

**Review by emadkaramad — DISMISSED** [review]

File: src/pages/login/index.tsx [hunk 2]
```diff
@@ -64,7 +68,8 @@ const handleEmailStage = async (
       props: {
         csrfToken,
         emailAddress,
-        emailError: "Enter a valid email address"
+        emailError: "Enter a valid email address",
+        serviceMessages: JSON.parse(JSON.stringify(serviceMessages))
```
Comment (daviesjamie) [inline src/pages/login/index.tsx:72]: What's going on here? Why do we need to `JSON.parse(JSON.stringify(...))`? That results in a no-op?

**Review by daviesjamie — CHANGES_REQUESTED** [review]

File: src/pages/faq.tsx (hunk 1 above)
Comment (MihaiMadeT) [reply src/pages/faq.tsx:40]: good spot, will remove

File: src/pages/login/index.tsx (hunk 2 above)
Comment (daviesjamie) [reply src/pages/login/index.tsx:72]: So the problem here is that we can't serialise a `Date()` object?

In that case, what is returned from `JSON.parse()` when you parse a date string? Can we use that as the type in our ServiceMessage instead?

Doing a `JSON.parse(JSON.stringify(...))` just feels like a bit of a workaround... 🤷

Comment (MihaiMadeT) [reply src/pages/login/index.tsx:72]: Sorry for the late reply:

It seems that this is an issue with JS and it does not like serializing anything but scalar types for performance reasons: https://github.com/vercel/next.js/issues/13209#issuecomment-633149650. We have an existing stringify line somewhere else in the code. JSON.parse() returns an object with keys and stuff.

**Review by daviesjamie — DISMISSED** [review]

**Review by daviesjamie — APPROVED** [review]

**Pull request merged**